src/cpp/server/server.cc

```diff
@@ -51,6 +51,15 @@
 
 namespace grpc {
 
+class DefaultGlobalCallbacks GRPC_FINAL : public Server::GlobalCallbacks {
+ public:
+  void PreSynchronousRequest(ServerContext* context) GRPC_OVERRIDE {}
+  void PostSynchronousRequest(ServerContext* context) GRPC_OVERRIDE {}
+};
+
+static DefaultGlobalCallbacks g_default_callbacks;
```

Comment:
Non-POD static variables can be problematic... 

I would not bother with a default impl that does nothing. Instead I would set `g_callbacks` to `nullptr` and test for it in the `Run` function.